tests/test_basic_functionality.py

```diff
@@ -34,7 +34,7 @@ def empty_node(layers: list[str]) -> ConfigNode:
 def full_node(layers_and_values: dict[str, str]) -> ConfigNode:
     n = ConfigNode(list(layers_and_values.keys()), name="test_node")
     for layer, value in layers_and_values.items():
-        n.update(value, layer, source=None)
+        n.update(value, layer, source="test_source")
```

Comment:
Why are you providing a source here? Do we have any tests with a None source, since the typing of sources is `Optional[str]`?